aws_lambda_powertools/utilities/data_classes/api_gateway_proxy_event.py

```diff
@@ -250,3 +258,6 @@ def path(self) -> str:
     def http_method(self) -> str:
         """The HTTP method used. Valid values include: DELETE, GET, HEAD, OPTIONS, PATCH, POST, and PUT."""
         return self.request_context.http.method
+
+    def header_serializer(self):
+        return HttpApiSerializer()
```

Comment:
Suggestion: change the name to `HttpApiHeaderSerializer` so it doesn't get confused by a newcomer later - e.g., `HttpApiSerializer` could mean something else in the far future.